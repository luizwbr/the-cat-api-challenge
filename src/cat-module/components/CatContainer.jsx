import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import CatImage from './CatImage.jsx';
import CatInformation from './CatInformation.jsx';

const PolaroidDiv = styled.div`
    background: #ffffff;
    display: inline-block;
    margin: 55px 75px 30px;
    padding: 15px 15px 30px;
    text-decoration: none;
    -webkit-box-shadow: 0 2px 3px rgba(0, 0, 0, .3);
    -moz-box-shadow: 0 2px 4px rgba(0, 0, 0, .3);
    box-shadow: 0 2px 4px rgba(0, 0, 0, .3);
    -webkit-transition: all .20s linear;
    -moz-transition: all .20s linear;
    transition: all .20s linear;
    z-index: 0;
    position: relative;

    :hover {
        -webkit-transform: scale(1.1);
        -moz-transform: scale(1.1);
        transform: scale(1.1);
        z-index: 10;
        -webkit-box-shadow: 0 5px 10px rgba(0, 0, 0, .7);
        -moz-box-shadow: 0 5px 10px rgba(0, 0, 0, .7);
        box-shadow: 0 5px 10px rgba(0, 0, 0, .7);
    }
`;

function CatContainer(props) {
    const { item } = props;

    return (
        <PolaroidDiv>
            <CatImage id={item?.id} url={item?.url} />
            <CatInformation item={item} />
        </PolaroidDiv>
    );
}

CatContainer.propTypes = {

    /**
     * item - the item to render
     */
    item: PropTypes.object.isRequired,
};

// eslint-disable-next-line import/no-unused-modules
export default CatContainer;
