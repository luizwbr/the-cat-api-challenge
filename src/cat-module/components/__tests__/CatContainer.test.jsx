import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import CatContainer from '../CatContainer.jsx';

const mockProps = {
    item: {
        breeds: [],
        id: '6rb',
        url: 'https://cdn2.thecatapi.com/images/6rb.gif',
        width: 450,
        height: 252,
    }
};

test('renders', () => {
    // eslint-disable-next-line max-len
    const { asFragment } = render(<CatContainer item={mockProps.item} />);

    expect(asFragment()).toMatchSnapshot();
});

