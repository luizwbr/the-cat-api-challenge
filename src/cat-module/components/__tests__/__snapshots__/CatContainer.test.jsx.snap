// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="sc-jqUVSM kiHODo"
  >
    <div
      class="sc-bczRLJ iHeqKj"
    >
      <img
        alt="cat-image-6rb"
        class="sc-gsnTZi dUmWvC"
        id="image-6rb"
        loading="lazy"
        src="https://cdn2.thecatapi.com/images/6rb.gif"
        width="400"
      />
    </div>
    <div
      class="sc-dkzDqf gfLwvs"
    >
      <div
        class="sc-hKMtZM jcFglv"
      >
        <h1
          class="sc-jSMfEi bJisEZ"
        >
          <span
            class="sc-iBkjds ehMTzg"
          >
            ID:
          </span>
          <span
            class="sc-ftvSup bYiFXU"
          >
            #6rb
          </span>
        </h1>
        <h2
          class="sc-gKXOVf bHhEWU"
        >
          <span
            class="sc-iBkjds ehMTzg"
          >
            Width:
          </span>
          <span
            class="sc-ftvSup bYiFXU"
          >
            450px
          </span>
        </h2>
        <h2
          class="sc-gKXOVf bHhEWU"
        >
          <span
            class="sc-iBkjds ehMTzg"
          >
            Height:
          </span>
          <span
            class="sc-ftvSup bYiFXU"
          >
            252px
          </span>
        </h2>
      </div>
      <div
        class="sc-eCYdqJ jhZexB"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-heart sc-papXJ lkxYlF"
          color="gray"
          data-icon="heart"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 190.9V185.1C0 115.2 50.52 55.58 119.4 44.1C164.1 36.51 211.4 51.37 244 84.02L256 96L267.1 84.02C300.6 51.37 347 36.51 392.6 44.1C461.5 55.58 512 115.2 512 185.1V190.9C512 232.4 494.8 272.1 464.4 300.4L283.7 469.1C276.2 476.1 266.3 480 256 480C245.7 480 235.8 476.1 228.3 469.1L47.59 300.4C17.23 272.1 .0003 232.4 .0003 190.9L0 190.9z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
